panel/panelpluginsmodel.h

```diff
@@ -48,52 +55,284 @@ class PanelPluginsModel : public QAbstractListModel
 
 private:
+    /*!
+     * \brief loadPlugins Loads all the Plugins.
+     * \param desktopDirs These directories are scanned for corresponding
+     * .desktop-files which are necessary to load the plugins.
+     */
     void loadPlugins(QStringList const & desktopDirs);
+    /*!
+     * \brief loadPlugin Loads a Plugin and connects signals and slots.
+     * \param desktopFile The desktop file that specifies how to load the
+     * Plugin.
+     * \param settingsGroup QString which specifies the settings group. This
+     * will only be redirected to the Plugin so that it knows how to read
+     * its settings.
+     * \return A QPointer to the Plugin that was loaded.
+     */
     QPointer<Plugin> loadPlugin(LXQt::PluginInfo const & desktopFile, QString const & settingsGroup);
+    /*!
+     * \brief findNewPluginSettingsGroup Creates a name for a new Plugin
+     * that is not yet present in the settings file. Whenever multiple
+     * instances of a single Plugin type are created, they have to be
+     * distinguished by this name.
+     *
+     * The first Plugin of a given type will be named like the type, e.g.
+     * "mainmenu". If a name is already present, this method tries to
+     * find a free name by appending increasing integers (starting with 2),
+     * e.g. "mainmenu2". If, for example, only "mainmenu2" exists because
+     * "mainmenu" was deleted, "mainmenu" would be returned. So, the method
+     * always finds the first suitable name that is not yet present in the
+     * settings file.
+     * \param pluginType Type of the Plugin.
+     * \return The created name for the Plugin.
+     */
     QString findNewPluginSettingsGroup(const QString &pluginType) const;
+    /*!
+     * \brief isIndexValid Checks if a given model index is valid for the
+     * underlying data (column 0, row lower than number of Plugins and
+     * so on).
+     */
     bool isIndexValid(QModelIndex const & index) const;
+    /*!
+     * \brief removePlugin Removes a given Plugin from the model.
+     */
     void removePlugin(pluginslist_t::iterator plugin);
 
+    /*!
+     * \brief mNamesKey The key to the settings-entry that stores the
+     * names of the Plugins in a panel. Set upon creation, passed as
+     * a parameter by the panel.
+     */
     const QString mNamesKey;
+    /*!
+     * \brief mPlugins Stores all the Plugins.
+     *
+     * mPlugins is a QList of elements while each element corresponds to a
+     * single Plugin. Each element is a QPair of a QString and a QPointer
+     * while the QPointer points to a Plugin.
+     *
+     * To access the elements, you can use indexing or an iterator on the
+     * list. For each element p, p.first is the name of the Plugin as it
+     * is used in the configuration files, p.second.data() is the Plugin.
+     *
+     * \sa pluginslist_t
+     */
     pluginslist_t mPlugins;
+    /*!
+     * \brief mPanel Stores a reference to the LXQtPanel.
```

Comment:
I'm not sure it's correct to say this. `mPanel` is a pointer to the object, does it store a *reference*? Anyway, I think the comments (although very valuable) make more sense for the functions, so the member objects should be left alone (unless one really needs explanation). Besides, in some cases (like this one), the comment is very obvious, and so it pollutes more than it helps.

Don't get me wrong, you're doing a great job. Besides all I said, GTM.